Sming/SmingCore/Network/TcpServer.cpp

```diff
@@ -129,7 +129,8 @@ err_t TcpServer::onAccept(tcp_pcb* clientTcp, err_t err)
 	connections.add(client);
 	debug_d("Opening connection. Total connections: %d", connections.count());
 
-	onClient((TcpClient*)client);
+	// @todo Dangerous up-cast - needs looking at
```

Comment:
I guess you will take a look at this before the PR is declared ready for merging?

Reply:
Here be demons. We have `virtual createClient()` returning a `TcpConnection*` which, in most cases, is actually a `TcpClient*`, except for `FTPServerConnection` which creates an `FTPServerConnection` inherited directly from `TcpConnection`. Logically, I'd expect `createClient` to `TcpClient*'. If that's the case it's a significant change to the framework and FTPServer needs looking at - there's a lot of extra stuff in `TcpClient` vs. `TcpConnection`.

If you have any thoughts on this, most welcome!

This'll need a separate PR, so I'll remove the comment.

Reply:
> This'll need a separate PR, so I'll remove the comment and put up with the 'C-style typecast' warning for now.

Perfect. Thanks!

Reply:
Just remove the comment so that I can merge the PR.

Reply:
@mikee47 Is the PR ready for merging?

Reply:
@slaff Yes, all done.